packages/snjs/lib/services/history/history_manager.ts

```diff
@@ -226,24 +317,95 @@ export class SNHistoryManager extends PureService {
    * Expands on a revision fetched via `remoteHistoryForItem` by getting a revision's
    * complete fields (including encrypted content).
    */
-  async fetchRemoteRevision(itemUuid: UuidString, entry: RevisionListEntry) {
-    const revision = (await this.apiService!.getRevision(
+  async fetchRemoteRevision(
+    itemUuid: UuidString,
+    entry: RevisionListEntry
+  ): Promise<HistoryEntry | undefined> {
+    const revision = (await this.apiService.getRevision(
       entry,
       itemUuid
     )) as SingleRevision;
     if ((revision as SingleRevisionResponse).error) {
       return undefined;
     }
-    const payload = CreateMaxPayloadFromAnyObject(revision as any, {
-      uuid: revision.item_uuid,
-    });
+    const payload = CreateMaxPayloadFromAnyObject(
+      (revision as unknown) as RawPayload,
+      {
+        uuid: revision.item_uuid,
+      }
+    );
     const encryptedPayload = CreateSourcedPayloadFromObject(
       payload,
       PayloadSource.RemoteHistory
     );
-    const decryptedPayload = await this.protocolService!.payloadByDecryptingPayload(
+    const decryptedPayload = await this.protocolService.payloadByDecryptingPayload(
       encryptedPayload
     );
-    return new ItemHistoryEntry(decryptedPayload);
+    if (decryptedPayload.errorDecrypting) {
+      return undefined;
+    }
+    return new HistoryEntry(decryptedPayload as SurePayload);
+  }
+
+  optimizeHistoryForItem(uuid: string): void {
+    /**
+     * Clean up if there are too many revisions. Note itemRevisionThreshold
+     * is the amount of revisions which above, call for an optimization. An
+     * optimization may not remove entries above this threshold. It will
```

Comment:
> An optimization may not remove entries above this threshold.

This would mean that on every item change, an optimize job gets started yes? Is this not a concern with regards to performance/chugging? Or do you expect this scenario to be unlikely?

Reply:
An optimize could be called on every change. It's not too resource intensive.